pkg/server/server.go

```diff
@@ -494,210 +564,225 @@ func (s *ImmuServer) GetBatch(ctx context.Context, kl *schema.KeyList) (*schema.
-	s.Logger.Debugf("consistency for index %d ", index.Index)
-	proof, err := s.Store.ConsistencyProof(*index)
+	ind, err := s.getDbIndexFromCtx(ctx, "Consistency")
 	if err != nil {
 		return nil, err
 	}
-	return proof, nil
+	return s.databases[ind].Consistency(index)
 }
 
 // ByIndex ...
 func (s *ImmuServer) ByIndex(ctx context.Context, index *schema.Index) (*schema.Item, error) {
 	s.Logger.Debugf("get by index %d ", index.Index)
-	item, err := s.Store.ByIndex(*index)
+	ind, err := s.getDbIndexFromCtx(ctx, "ByIndex")
 	if err != nil {
 		return nil, err
 	}
-	return item, nil
+	return s.databases[ind].ByIndex(index)
 }
 
 // ByIndexSV ...
 func (s *ImmuServer) ByIndexSV(ctx context.Context, index *schema.Index) (*schema.StructuredItem, error) {
 	s.Logger.Debugf("get by index %d ", index.Index)
-	item, err := s.Store.ByIndex(*index)
+	ind, err := s.getDbIndexFromCtx(ctx, "ByIndexSV")
 	if err != nil {
 		return nil, err
 	}
-	sitem, err := item.ToSItem()
+	item, err := s.databases[ind].ByIndex(index)
 	if err != nil {
 		return nil, err
 	}
-	return sitem, nil
+	return item.ToSItem()
 }
 
 // BySafeIndex ...
 func (s *ImmuServer) BySafeIndex(ctx context.Context, sio *schema.SafeIndexOptions) (*schema.SafeItem, error) {
 	s.Logger.Debugf("get by safeIndex %d ", sio.Index)
-	item, err := s.Store.BySafeIndex(*sio)
+	ind, err := s.getDbIndexFromCtx(ctx, "BySafeIndex")
 	if err != nil {
 		return nil, err
 	}
-	return item, nil
+	return s.databases[ind].BySafeIndex(sio)
 }
 
 // History ...
 func (s *ImmuServer) History(ctx context.Context, key *schema.Key) (*schema.ItemList, error) {
 	s.Logger.Debugf("history for key %s ", string(key.Key))
-	list, err := s.Store.History(*key)
+	ind, err := s.getDbIndexFromCtx(ctx, "History")
 	if err != nil {
 		return nil, err
 	}
-	return list, nil
+	return s.databases[ind].History(key)
 }
 
 // HistorySV ...
 func (s *ImmuServer) HistorySV(ctx context.Context, key *schema.Key) (*schema.StructuredItemList, error) {
 	s.Logger.Debugf("history for key %s ", string(key.Key))
-
-	list, err := s.Store.History(*key)
+	ind, err := s.getDbIndexFromCtx(ctx, "HistorySV")
 	if err != nil {
 		return nil, err
 	}
-
-	slist, err := list.ToSItemList()
+	list, err := s.databases[ind].History(key)
 	if err != nil {
 		return nil, err
 	}
-	return slist, err
+	return list.ToSItemList()
 }
 
 // Health ...
-func (s *ImmuServer) Health(context.Context, *empty.Empty) (*schema.HealthResponse, error) {
-	health := s.Store.HealthCheck()
-	s.Logger.Debugf("health check: %v", health)
-	return &schema.HealthResponse{Status: health}, nil
+func (s *ImmuServer) Health(ctx context.Context, e *empty.Empty) (*schema.HealthResponse, error) {
+	//s.Logger.Debugf("health check: %v", health)
+	ind, _ := s.getDbIndexFromCtx(ctx, "Health")
+
+	if ind < 0 { //probably immuclient hasn't logged in yet
+		return s.databases[0].Health(e)
+	}
+	fmt.Println("sdfsdfsdfsdfsdfsdfsdf", ind)
```

Comment:
@Gjergj if this is not used anymore it should be deleted.

Reply:
Yes, it's already removed in the next commit auditor.